spacy_llm/models/hf/base.py

```diff
@@ -27,12 +27,47 @@ def __init__(
         inference_config (Dict[Any, Any]): HF config for model run.
         """
         self._name = name if self.hf_account in name else f"{self.hf_account}/{name}"
-        self._config_init, self._config_run = self.compile_default_configs()
+        default_cfg_init, default_cfg_run = self.compile_default_configs()
+        self._config_init, self._config_run = default_cfg_init, default_cfg_run
+
         if config_init:
             self._config_init = {**self._config_init, **config_init}
         if config_run:
             self._config_run = {**self._config_run, **config_run}
 
+        # `device` and `device_map` are conflicting arguments - ensure they aren't both set.
+        # Case 1: we have a CUDA GPU (and hence device="cuda:0" by default), but device_map is set by user.
+        if config_init:
+            if "device" in default_cfg_init and "device_map" in config_init:
+                self._config_init.pop("device")
+            # Case 2: we don't have a CUDA GPU (and hence "device_map=auto" by default), but device is set by user.
+            if "device_map" in default_cfg_init and "device" in config_init:
+                self._config_init.pop("device_map")
+            # Case 3: both explicitly set by user.
+            if "device" in config_init and "device_map" in config_init:
+                warnings.warn(
+                    "`device` and `device_map` are conflicting arguments - don't set both. Dropping argument "
+                    "`device`."
+                )
+                if "device" in self._config_init:
+                    self._config_init.pop("device")
```

Comment:
I would swap these cases and make the current Case 3 the first `if` check. The reason is that if the user has both defined in their `config_init`, the behaviour of which is removed shouldn't depend on previous checks against the `default_cfg_init`, and we'd probably want to warn explicitely about the conflict so they understand that one might be removed. Also by having it as first condition, you can remove the extra check `if "device" in self._config_init`. The other two can then be `elif`'s.

Reply:
Done in https://github.com/explosion/spacy-llm/pull/359/commits/17a6402c2f2a230a046a95a59637a4e0fd30e363.